src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -994,11 +995,18 @@ def _create_x509_extension(self, handlers, extension):
             value = _encode_asn1_str_gc(self, extension.value.value)
             return self._create_raw_x509_extension(extension, value)
         elif isinstance(extension.value, x509.TLSFeature):
-            asn1 = _Integers([x.value for x in extension.value]).dump()
+            asn1 = encode_der(
+                SEQUENCE,
+                *[
+                    encode_der(INTEGER, encode_der_integer(x.value))
+                    for x in extension.value
+                ]
+            )
             value = _encode_asn1_str_gc(self, asn1)
             return self._create_raw_x509_extension(extension, value)
         elif isinstance(extension.value, x509.PrecertPoison):
-            asn1 = asn1crypto.core.Null().dump()
+            # The contents are an ASN.1 NULL.
+            asn1 = b'\x05\x00'
```

Comment:
Should this be `encode_der(NULL)`? It's slightly silly since it constant folds, on the other hand it makes the code self-documenting.

Reply:
Done.

Reply:
Avoided hardcoding things in decode_asn1.py too, which is actually shorter anyway.